### PR TITLE
test: cypress - updated tests for List page js to ts migration

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
- migration of flaky list tests from js to ts

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_defaultSelectItem_spec.ts [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    deployMode.NavigateBacktoEditor();
+  });
+});
+
+describe("List widget v2 Reset List widget and Refresh Data", () => {
+  it("1. Setup List Widget", () => {
+    entityExplorer.SelectEntityByName("List1");
+    // Here updateLayout call is not happenign hence cant use UpdatePropertyFieldValue()
+    //propPane.UpdatePropertyFieldValue("Default selected item", "4");
+    propPane.TypeTextIntoField(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_defaultSelectItem_spec.ts:170]: Use `UpdatePropertyFieldValue` with `toValidateNetworkCall` parameter set to `false`.

**Review by Aishwarya-U-R — COMMENTED** [review]
In PR description - give the proper names of specs/folder migrated in this PR. The current one line does not suffice.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+const dsl = require("../../../../../fixtures/listdsl.json");
+
+describe("Container Widget Functionality", function () {
+  const items = JSON.parse(dsl.dsl.children[0].listData);
+
+  before(() => {
+    agHelper.AddDsl("listdsl");
+  });
+
+  it("1. List-Unckeck Visible field Validation", function () {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts:19]: Give proper `it` description. `Validate Visibility via Toggle`

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    //Check the disableed checkbox and Validate
+    propPane.TogglePropertyState("Visible", "On");
+    deployMode.DeployApp();
+    agHelper.AssertElementVisible(
+      locators._widgetInDeployed(draggableWidgets.LIST),
+    );
+    deployMode.NavigateBacktoEditor();
+  });
+
+  it("2. Toggle JS - List-Unckeck Visible field Validation", function () {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts:41]: Give proper `it` description. `Validate Visibility via JS Mode`

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+  agHelper,
+  locators,
+  propPane,
+  deployMode,
+  appSettings,
+  draggableWidgets,
+} from "../../../../../support/Objects/ObjectsCore";
+const dsl = require("../../../../../fixtures/listdsl.json");
+
+describe("Container Widget Functionality", function () {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts:12]: This spec is Container or List?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    // Verify the length of list
+    agHelper.AssertElementLength(
+      locators._widgetInDeployed(draggableWidgets.CONTAINER),
+      2,
+    );
+    // Clear item spacing
+    propPane.UpdatePropertyFieldValue("Item Spacing (px)", "");
+    cy.wait(2000);
+    // Close property pane
+    agHelper.GetNClick(appSettings.locators._canvas);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts:107]: Here too?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+      propPane._propertyText,
+      items[1].first_name,
+      "contain.text",
+      1,
+    );
+  });
+
+  it("4. doesn't alter the no of items present when invalid item spacing is entered", () => {
+    // Open Property pane
+    entityExplorer.SelectEntityByName("List1", "Widgets");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts:94]: List1 properties should be selected here from previous test already. Is this needed?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+    // Close property pane
+    agHelper.GetNClick(appSettings.locators._canvas);
+  });
+
+  it("5. checks button action", function () {
+    // Open property pane
+    entityExplorer.ExpandCollapseEntity("Widgets");
+    entityExplorer.ExpandCollapseEntity("List1");
+    entityExplorer.ExpandCollapseEntity("Container1");
+    entityExplorer.SelectEntityByName("Button1");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts:115]: Until container - its already expanded. Is this chain again needed here?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_DataIdentifierProperty_spec.ts [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+        // we get a list of jQuery elements
+        // convert the jQuery object into a plain array
+        return (
+          Cypress.$.makeArray($el)
+            // extract inner text from each
+            .map((el: any) => el.innerText)
+        );
+      })
+      .should("deep.equal", keys);
+  });
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_DataIdentifierProperty_spec.ts:75]: Use `propPane.AssertPropertiesDropDownValues`. I will merge this method in sometime.

File: app/client/cypress/support/Pages/PropertyPane.ts [hunk 9]
```diff
@@ -527,4 +530,9 @@ export class PropertyPane {
   public GetSelectedItemText(property: string) {
     return this.agHelper.GetText(this._propPaneSelectedItem(property));
   }
+
+  public SelectColorFromColorPicker(property: string, colorOffset = -15) {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/PropertyPane.ts:534]: Why is colorOffset defaulted to `-15`?

File: app/client/cypress/support/Objects/CommonLocators.ts [hunk 10]
```diff
@@ -259,11 +259,18 @@ export class CommonLocators {
   _pagination = ".rc-pagination";
   _controlOption = ".t--property-control-options";
   _canvasBody = "[data-testid='div-selection-0']";
+  _itemContainerWidget = ".t--widget-containerwidget div.style-container";
+  _paginationButton = ".rc-pagination-item";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Objects/CommonLocators.ts:263]: Seeing duplicate elements in `Table.ts `also. Remove from here.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_defaultSelectItem_spec.ts [hunk 11]
```diff
@@ -0,0 +1,280 @@
+import {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_defaultSelectItem_spec.ts:1]: Follow one naming convention. Remove `Listv2` from all spec names.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_DataIdentifierProperty_spec.ts (hunk 8 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_DataIdentifierProperty_spec.ts:75]: This is merged & can be used.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts (hunk 2 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts (hunk 3 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+    agHelper.AssertElementLength(
+      locators._widgetInDeployed(draggableWidgets.CONTAINER),
+      2,
+    );
+    //checks currentItem binding
+    // Open property pane
+    entityExplorer.ExpandCollapseEntity("Widgets");
+    entityExplorer.ExpandCollapseEntity("List1");
+    entityExplorer.ExpandCollapseEntity("Container1");
+    entityExplorer.SelectEntityByName("Text1");
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts:72]: done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts [hunk 13]
```diff
@@ -0,0 +1,245 @@
...
+    propPane.UpdatePropertyFieldValue("Text", `{{currentItem.first_name}}`);
+    agHelper.GetNClick(appSettings.locators._canvas);
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts]: removed

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts [hunk 14]
```diff
@@ -0,0 +1,245 @@
...
+    // Update an invalid value to item spacing
+    propPane.UpdatePropertyFieldValue("Item Spacing (px)", "-");
+    // Verify the length of list
+    agHelper.AssertElementLength(
+      locators._widgetInDeployed(draggableWidgets.CONTAINER),
+      2,
+    );
+    // Clear item spacing
+    propPane.UpdatePropertyFieldValue("Item Spacing (px)", "");
+    cy.wait(2000);
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts (hunk 5 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts (hunk 6 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts:91]: updateed

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts (hunk 7 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/List/List4_1_spec.ts]: updated

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_DataIdentifierProperty_spec.ts [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+
+    entityExplorer.SelectEntityByName("List1");
+
+    agHelper.GetNAssertContains(
+      `${locators._propertyControl}dataidentifier`,
+      "Please select an option",
+    );
+  });
+
+  it("2. shows list of keys present in list data", () => {
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_DataIdentifierProperty_spec.ts]: donne

File: app/client/cypress/support/Pages/PropertyPane.ts (hunk 9 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/PropertyPane.ts]: udpated

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_defaultSelectItem_spec.ts (hunk 11 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_defaultSelectItem_spec.ts]: removed

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_DataIdentifierProperty_spec.ts (hunk 8 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Listv2_DataIdentifierProperty_spec.ts]: updated

File: app/client/cypress/support/Objects/CommonLocators.ts (hunk 10 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Objects/CommonLocators.ts:263]: removed duplicates

**Comment by Aishwarya-U-R** [comment]
> 1. TableV2_GeneralProperty_spec

@NandanAnantharamu - fix this flaky test in another open PR.

File: app/client/cypress/support/Pages/Table.ts [hunk 16]
```diff
@@ -733,4 +782,20 @@ export class Table {
         });
     });
   }
+
+  public NavigateToSpecificPage(pageNumber: number) {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/Table.ts:786]: Why is this needed? I dont see any reference.

File: app/client/cypress/support/Pages/Table.ts [hunk 17]
```diff
@@ -733,4 +782,20 @@ export class Table {
...
+    this.agHelper.GetNClick(`${this.locator._paginationItem(pageNumber)}`);
+    this.agHelper
+      .GetText(this.locator._listActivePage, "text")
+      .then(($newPageNo) => expect(Number($newPageNo)).to.eq(pageNumber));
+  }
+
+  //This method is used to navigate forward using ">" button and backward "<"
+  public NavigateToPageUsingButton(movement: string, pageNumber: number) {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/Table.ts:794]: What page is this? Is this list page you mean? I thought App-Page.
Rename is accordingly.

Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/Table.ts:794]: done

File: app/client/cypress/support/Pages/Table.ts (hunk 16 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/Table.ts:786]: yeah this was needed for another test which is there in another open PR, i will remove this will be using it over there, this is to navigate to specific page unlike next page

File: app/client/cypress/support/Pages/PropertyPane.ts [hunk 18]
```diff
@@ -518,9 +521,17 @@ export class PropertyPane {
     this.agHelper.GetNClick(this._addAction(property), 0, true);
   }
 
-  public SelectPlatformFunction(eventName: string, dropdownValue: string) {
+  public SelectPlatformFunction(
+    eventName: string,
+    dropdownValue: string,
+    force = true,
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/PropertyPane.ts:527]: i have made it by default true, so caller i am not passing any value. This was needed for a test in List4_1_spec

File: app/client/cypress/support/Pages/Table.ts [hunk 19]
```diff
@@ -733,4 +782,13 @@ export class Table {
         });
     });
   }
+
+  //This method is used to navigate forward using ">" button and backward "<"
+  public NavigateToPageUsingButtonFrList(movement: string, pageNumber: number) {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/Table.ts:787]: `NavigateToPageUsingButton_List` - name similar to other existing methods

File: app/client/cypress/support/Pages/Table.ts [hunk 20]
```diff
@@ -783,15 +783,8 @@ export class Table {
     });
   }
 
-  public NavigateToSpecificPage(pageNumber: number) {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/Table.ts]: Bring this method back. It might be needed. But name it appropriately.

File: app/client/cypress/support/Pages/PropertyPane.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/PropertyPane.ts:527]: Why does it have to be defaulted to true at method level - for all `SelectPlatformFunction` calls()? Make change only at the place you need now.

File: app/client/cypress/support/Pages/Table.ts (hunk 20 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/Table.ts]: done

File: app/client/cypress/support/Pages/Table.ts (hunk 19 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/Table.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DataIdentifierProperty_spec.ts [hunk 21]
```diff
@@ -0,0 +1,209 @@
...
+      3,
+    );
+  });
+
+  it("3. enabling the JS mode, it should prefill with currentItem", () => {
+    propPane.ToggleJSMode("Data Identifier", true);
+    agHelper
+      .GetElement(locators._existingFieldTextByName("Data Identifier"))
+      .find(".CodeMirror .CodeMirror-code")
+      .contains(`{{ currentItem["id"] }}`);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DataIdentifierProperty_spec.ts:74]: Use - `propPane.ValidateJSFieldValue`

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DataIdentifierProperty_spec.ts [hunk 22]
```diff
@@ -0,0 +1,209 @@
...
+describe("List v2 - Data Identifier property", () => {
+  beforeEach(() => {
+    agHelper.RestoreLocalStorageCache();
+  });
+
+  afterEach(() => {
+    agHelper.SaveLocalStorageCache();
+  });
+
+  it("1. is present in the property pane", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DataIdentifierProperty_spec.ts:42]: What `is` present?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DataIdentifierProperty_spec.ts [hunk 23]
```diff
@@ -0,0 +1,209 @@
...
+    agHelper.AddDsl("Listv2/simpleList");
+
+    entityExplorer.SelectEntityByName("List1");
+
+    agHelper.GetNAssertContains(
+      `${locators._propertyControl}dataidentifier`,
+      "Please select an option",
+    );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DataIdentifierProperty_spec.ts:50]: use `propPane.AssertPropertiesDropDownCurrentValue`

File: app/client/cypress/support/Pages/Table.ts [hunk 24]
```diff
@@ -693,17 +708,51 @@ export class Table {
...
+          "true",
+        );
+
+      if (checkNoNextPage)
+        cy.get(this._listNextPage).should("have.attr", "aria-disabled", "true");
+      else
+        cy.get(this._listNextPage).should(
+          "have.attr",
+          "aria-disabled",
+          "false",
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/Table.ts:734]: USe `agHelper.AssertAttribute`

File: app/client/cypress/support/Objects/CommonLocators.ts [hunk 25]
```diff
@@ -260,12 +260,20 @@ export class CommonLocators {
...
+  _firstChild = ":first-child";
   _adsV2Content = ".ads-v2__content";
   _adsV2CollapsibleHeader = ".ads-v2-collapsible__header";
   _adsV2Text = ".ads-v2-text";
   _svg = "svg";
   _imgWidgetInsideList = `//div[@data-testid='styledImage']//img`;
-
+  _containerWidget = "[type='CONTAINER_WIDGET']";
+  _listActivePage = ".t--widget-listwidgetv2 .rc-pagination-item-active";
+  _paginationItem = (value: number) => `.rc-pagination-item-${value}`;
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Objects/CommonLocators.ts:273]: Move locators to Table.ts

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/PropertyPane_spec.ts [hunk 26]
```diff
@@ -0,0 +1,76 @@
...
+  entityExplorer,
+  draggableWidgets,
+  deployMode,
+  propPane,
+} from "../../../../../support/Objects/ObjectsCore";
+
+describe("List widget V2 PropertyPane", () => {
+  it("1. Validate isVisible", () => {
+    entityExplorer.DragDropWidgetNVerify("listwidgetv2", 300, 300);
+    entityExplorer.SelectEntityByName("List1");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/PropertyPane_spec.ts:13]: During `DragDropWidgetNVerify`, `SelectEntityByName` is unnecessary

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/PropertyPane_spec.ts [hunk 27]
```diff
@@ -0,0 +1,76 @@
+import {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/PropertyPane_spec.ts:1]: Why is js of this spec not removed?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Event_Bindings_spec.ts [hunk 28]
```diff
@@ -0,0 +1,70 @@
...
+    // Enter text in the child list widget's text input in first row
+    agHelper
+      .GetElement(locators._widgetByName("Input2"))
+      .find("input")
+      .type("inner input");
+
+    // click the button on inner list 1st row.
+    agHelper
+      .GetElement(locators._widgetByName("Button3"))
+      .find("button")
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/Event_Bindings_spec.ts:39]: Reason why `TypeText` & `ClickButton` not used?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DefaultSelectItem_spec.ts [hunk 29]
```diff
@@ -0,0 +1,280 @@
...
+    verifyDefaultItem();
+  });
+
+  it("2. Reset List Widget", () => {
+    // Select a new List Item on another page
+    agHelper.GetNClick(
+      `${locators._widgetByName("List1")} ${locators._paginationItem(1)}`,
+      0,
+      true,
+    );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DefaultSelectItem_spec.ts:186]: Why was new method not used here? - `NavigateToSpecificPage_List`?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DefaultSelectItem_spec.ts [hunk 30]
```diff
@@ -0,0 +1,280 @@
...
+      .GetElement(
+        `${locators._widgetByName("SelectedItemView")} ${
+          propPane._propertyText
+        }`,
+      )
+      .then((val) => {
+        const data = JSON.parse(val.text());
+        cy.waitUntil(() => cy.wrap(data?.Text11?.text).should("equal", "4"));
+      }),
+  );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DefaultSelectItem_spec.ts:28]: Why was `GetText` not used here?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DefaultSelectItem_spec.ts [hunk 31]
```diff
@@ -0,0 +1,280 @@
...
+          const data = JSON.parse(val.text());
+          cy.wrap(data?.id).should("deep.equal", 1);
+        }),
+    );
+
+    agHelper.GetNClick(
+      `${locators._widgetByName("ResetWidget")} button`,
+      0,
+      true,
+    );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/ListV2/DefaultSelectItem_spec.ts:213]: Why `ClickButton` is not used?

**Comment by Aishwarya-U-R** [comment]
@sarojsarab - please help review [last commit](https://github.com/appsmithorg/appsmith/pull/25409/commits/bedd4274461d54fb33f2f799a93e2f38d921a27d) once.

**Comment by Aishwarya-U-R** [comment]
@NandanAnantharamu - cypress/e2e/Regression/ClientSide/DynamicHeight/Container_collapse_undo_redoSpec.ts is failing locally also hence have skipped now. Fix & unskip it.

File: app/client/cypress/support/Pages/AggregateHelper.ts [hunk 32]
```diff
@@ -761,10 +761,36 @@ export class AggregateHelper extends ReusableHelper {
   public TypeText(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/AggregateHelper.ts:761]: @Parthvi12 - below is the change

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/ActionSelector_JsToNonJSMode_2_spec.ts [hunk 33]
```diff
@@ -503,8 +503,7 @@ describe("JS to non-JS mode in Action Selector", () => {
     agHelper.TypeText(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/ActionExecution/ActionSelector_JsToNonJSMode_2_spec.ts:503]: At caller end, this will be change @Parthvi12

File: app/client/cypress/support/Pages/AggregateHelper.ts (hunk 32 above)
Comment (Parthvi12) [reply app/client/cypress/support/Pages/AggregateHelper.ts:761]: Okay, should not affect EE, but let's monitor.
